From 35b42898a9528c4ca1c1ba8dc081d12d6d6f8047 Mon Sep 17 00:00:00 2001
From: Mike Sul <mike.sul@foundries.io>
Date: Sat, 3 Jul 2021 20:37:08 -0300
Subject: [PATCH] ostree-fetcher-curl: set a timeout for an overall request
 processing

Signed-off-by: Mike Sul <mike.sul@foundries.io>
Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
Signed-off-by: Jose Quaresma <jose.quaresma@foundries.io>
---
 src/libostree/ostree-fetcher-curl.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/libostree/ostree-fetcher-curl.c b/src/libostree/ostree-fetcher-curl.c
index 522eacfb..2d058e84 100644
--- a/src/libostree/ostree-fetcher-curl.c
+++ b/src/libostree/ostree-fetcher-curl.c
@@ -933,6 +933,15 @@ initiate_next_curl_request (FetcherRequest *req,
   rc = curl_easy_setopt (req->easy, CURLOPT_PROGRESSDATA, task);
   g_assert_cmpint (rc, ==, CURLM_OK);
 
+  /* set a request timeout, make sure it's not 0, otherwise an overall ostree pull session might hang */
+  long curl_timeout = 0L;
+  const char* curl_timeout_str = g_getenv ("OSTREE_CURL_TIMEOUT");
+  if (curl_timeout_str != NULL)
+    curl_timeout = atoi(curl_timeout_str);
+  if (curl_timeout == 0)
+    curl_timeout = 780L;
+  curl_easy_setopt (req->easy, CURLOPT_TIMEOUT, curl_timeout);
+
   CURLMcode multi_rc = curl_multi_add_handle (self->multi, req->easy);
   g_assert (multi_rc == CURLM_OK);
 }
